src/ffi_bindings/mod.rs

```diff
@@ -30,10 +30,14 @@ pub extern "C" fn detect_sql_injection(
             Err(_) => return 2, // Return error code if invalid UTF-8
         };
 
-        if detect_sql_injection_str(query_str, userinput_str, dialect) {
+        let detection_results = detect_sql_injection_str(query_str, userinput_str, dialect);
+        if let DetectionReason::FailedToTokenizeQuery = detection_results.reason {
+            // make a special exception for failing to tokenize query (report code 3)
+            return 3;
+        }
+        if detection_results.detected {
```

Comment:
Perhaps we can extract a function to transform result in a code? So that we can use that later for the wasm API?

Reply:
Discussed: we'll do it later when we change WASM